rules/validator.go

```diff
@@ -178,6 +189,25 @@ func (v *validator) validateRollupTags(ruleName string, tags []string) error {
 			return fmt.Errorf("rollup rule %s does not have required rollup tag: %s, provided rollup tags are %v", ruleName, requiredTag, tags)
 		}
 	}
+
+	return nil
+}
+
+func validateChars(str string, invalidChars []rune) error {
+	if len(invalidChars) == 0 {
+		return nil
+	}
+
+	// Validate that given string doesn't contain an invalid character
+	chars := make(map[rune]struct{}, len(invalidChars))
+	for _, invalidChar := range invalidChars {
+		chars[invalidChar] = struct{}{}
+	}
+	for _, char := range str {
+		if _, exists := chars[char]; exists {
+			return fmt.Errorf("%d is an invalid character", char)
```

Comment:
nit: perhaps it should be `"%s contains invalid character %v", char`? Also would be good to mention the rule name here as well to give more context around the error.

Reply:
Sgtm regarding "%s contains invalid character %v" . Regarding the rule name, since this validator is a simple util that could be used anywhere, I wanted to avoid passing in values that contain knowledge of specific things like rule name. The validators that do use util would append this error to their own message. Ie. this function validates that if a string contains invalid chars and bubbles that up. Happy to change if you think it would be better to add rule name however.

Reply:
Yeah fair point, my concern was that the error message will be bubbled back to the frontend, and when the end user sees this message, having a rule name might help them identify the root cause of the issue more quickly. But I see that the caller reformats the error to include the rule name, works for me.